src/AWS.Deploy.Common/Recipes/OptionSettingItem.ValueOverride.cs

```diff
@@ -150,7 +148,7 @@ public void SetValueOverride(object valueOverride)
                 {
                     if (deserialized.TryGetValue(childOptionSetting.Id, out var childValueOverride))
                     {
-                        childOptionSetting.SetValueOverride(childValueOverride);
+                        childOptionSetting.SetValue(childValueOverride);
```

Comment:
Something to think about. Having the set call calling directly to the child skips any validation that could be happening in the `IOptionSettingHandler`. I assume at this point if a setting is being done at the object level we are assuming the whole object has been validated. This might not always be true and could be the source of future problems.

I'm fine revising this later to get this initial PR to unblock others but maybe making a comment here for future work we need to do.

Reply:
Perfect, thanks